src/utils/DepositUtils.ts

```diff
@@ -55,15 +55,27 @@ export async function queryHistoricalDepositForFill(
   let deposit: DepositWithBlock, cachedDeposit: Deposit | undefined;
   if (cache) {
     cachedDeposit = await getDepositInCache(getDepositKey(fill), cache);
+    // We only want to warn and remove the cached deposit if it
+    //    A: exists
+    //    B: is not formed correctly
+    if (isDefined(cachedDeposit) && !isDepositFormedCorrectly(cachedDeposit)) {
+      spokePoolClient.logger.warn({
+        at: "[SDK]:DepositUtils#queryHistoricalDepositForFill",
+        message: "Cached deposit was not formed correctly, removing from cache",
+        fill,
+        cachedDeposit,
+      });
+      // By setting this value to undefined, we eventually have to pull
+      // the deposit from our spoke pool client. Because this new deposit
+      // is formed correctly, we will cache it below.
+      cachedDeposit = undefined;
```

Comment:
@mrice32 @pxrl @nicholaspai added a comment to highlight using the existing logic